File: Callback/simple_example.py
'''
@Description: 回调函数python版本示例
@Author: 陆海
@Date: 2019-12-17 17:56:50
@LastEditTime: 2019-12-17 18:08:21
@LastEditors: Please set LastEditors
'''

def double(x):
    return x*2

def quadruple(x):
    return x*4

def getOddNumber(k, getNumber):
    return 1 + getNumber(k)

def main():
    k = 1
    
    i = getOddNumber(k, double)
    print(i)

    i = getOddNumber(k, quadruple)
    print(i)

    i = getOddNumber(k, lambda x:8*x)
    print(i)

if __name__ == "__main__":
    main()

File: Callback/simple_example.c
/*
 * @Description: 回调函数C语言版本
 * @Author: 陆海
 * @Date: 2019-12-17 17:55:33
 * @LastEditTime: 2019-12-17 18:01:00
 * @LastEditors: Please set LastEditors
 */
#include <stdio.h>

typedef void (*callbackexample_t) (void);

void callback1(void)
{
    printf("callback test1\n");
}

void callback2(void)
{
    printf("callback test2\n");
}

int main(void)
{
    callbackexample_t callback;

    callback = callback1;

    callback();

    callback = callback2;

    callback();
    
    return 0;
}